File: leetcode/1052-grumpy-bookstore-owner/maxSatisfied_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"reflect"
	"testing"

	"github.com/edram/leetcode/util"
)

type MaxSatisfiedCase struct {
	Input struct {
		Customers []int
		Grumpy    []int
		Minutes   int
	}

	Output int
}

func TestMaxSatisfied(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MaxSatisfiedCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("input: %v ==> %v \n", testCase.Input, testCase.Output)

		result := maxSatisfied(testCase.Input.Customers, testCase.Input.Grumpy, testCase.Input.Minutes)
		if !reflect.DeepEqual(result, testCase.Output) {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/1052-grumpy-bookstore-owner/maxSatisfied.test.ts
import cases from "./cases.json";
import maxSatisfied from "./maxSatisfied";

test("grumpy bookstore owner", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = maxSatisfied(
      testCase.input.customers,
      testCase.input.grumpy,
      testCase.input.minutes
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );
    expect(result).toEqual(testCase.output);
  });
});
